workflow/executor/executor.go

```diff
@@ -211,6 +211,12 @@ func (we *WorkflowExecutor) LoadArtifacts() error {
 			if err != nil {
 				return errors.InternalWrapError(err)
 			}
+			err = filepath.Walk(artPath, func(path string, f os.FileInfo, err error) error {
+				return os.Chmod(path, os.FileMode(*art.Mode))
```

Comment:
what do I do if I want different permission for directories and files?

Reply:
Hadn't considered that use case actually. My understanding is that this feature wasn't used too much and the issue this PR is related to denoted a bug.

You guys have a much better understanding of what users will expect at this point so do you think it's worth adding something to the API to allow some sort of `recursive mode` implementation? Happy to add it if we think it's best that way! 😄

Reply:
I think we may want to add a boolean flag for recursion (e.g. `art.RecurseMode`). Let. me chat with the core team.

Reply:
Ok. Can we please request a new flag so you can opt in to this feature? At the moment it may break peoples workflows because it changes the permission of files that were not changed before.

Reply:
On it! Thanks for the quick response! 😄